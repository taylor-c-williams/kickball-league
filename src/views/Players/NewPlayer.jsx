import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import PlayerNewForm from '../../components/Players/PlayerNewForm';
import { createPlayer } from '../../services/players';

export default function NewPlayer() {
	const [name, setName] = useState('');
	const [position, setPosition] = useState('');
	const [teamId, setTeamId] = useState('');
	const history = useHistory();

	const handleSubmit = async (e) => {
		e.preventDefault();
		const res = await createPlayer({ name, position, teamId });
		history.push(`/players/${res[0].id}`);
	};
	return (
		<div>
			<h1>Add New Player</h1>
			<PlayerNewForm
				name={name}
				position={position}
				teamId={teamId}
				handleSubmit={handleSubmit}
				setName={setName}
				setPosition={setPosition}
				setTeamId={setTeamId}
			/>
		</div>
	);
}
