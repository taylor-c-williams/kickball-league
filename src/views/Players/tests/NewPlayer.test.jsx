import {
	waitForElementToBeRemoved,
	render,
	screen,
} from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import NewPlayer from '../NewPlayer';

it('should render accurately the create new player view and component', async () => {
	const container = render(
		<MemoryRouter initialEntries={['/newPlayer']}>
			<Route path="/newPlayer">
				<NewPlayer />
			</Route>
		</MemoryRouter>
	);

	await waitForElementToBeRemoved(() => screen.queryByText(/Loading.../i));
	const textEntry = await screen.findByText(/Position/i);

	expect(container).toMatchSnapshot();
	expect(textEntry).toBeInTheDocument();
});
