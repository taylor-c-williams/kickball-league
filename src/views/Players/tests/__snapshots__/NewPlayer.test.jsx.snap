// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render accurately the create new player view and component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Add New Player
        </h1>
        <div>
          <fieldset>
            <legend>
              New Player
            </legend>
            <form>
              <label
                for="name"
              >
                Player Name:
              </label>
              <input
                id="name"
                name="name"
                type="text"
                value=""
              />
              <label
                for="position"
              >
                Position
              </label>
              <input
                id="position"
                name="position"
                type="text"
                value=""
              />
              <label
                for="teamId"
              >
                Team
              </label>
              <select
                aria-label="teamId"
              >
                <option
                  value="1"
                >
                  team name
                </option>
                <option
                  value="8"
                >
                  ꓘool ꓘidz
                </option>
              </select>
              <input
                aria-label="Add New Player"
                type="submit"
                value="Add New Player"
              />
            </form>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Add New Player
      </h1>
      <div>
        <fieldset>
          <legend>
            New Player
          </legend>
          <form>
            <label
              for="name"
            >
              Player Name:
            </label>
            <input
              id="name"
              name="name"
              type="text"
              value=""
            />
            <label
              for="position"
            >
              Position
            </label>
            <input
              id="position"
              name="position"
              type="text"
              value=""
            />
            <label
              for="teamId"
            >
              Team
            </label>
            <select
              aria-label="teamId"
            >
              <option
                value="1"
              >
                team name
              </option>
              <option
                value="8"
              >
                ꓘool ꓘidz
              </option>
            </select>
            <input
              aria-label="Add New Player"
              type="submit"
              value="Add New Player"
            />
          </form>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
